### PR TITLE
created tests for API routes

### PR DESCRIPTION
**Pull request by redperezz**
mongo.test.js - just included the done() callback inside the test

game_model.js - put a check to see if game that was searched is defined, thrown error if not

server/app.js - moved app.listen to another file (server/server.js), I will delete that commented out port variable

games.test.js - tests for single game and multiple game routes 

routes/games.js - defined proper error codes for response and testing purposes

server/server.js - contains app.listen

### CODE REVIEW

File: globalConfig.json [hunk 1]
```diff
@@ -1 +1 @@
-{"mongoUri":"mongodb://127.0.0.1:56096/f03ccc31-5cdc-4612-8799-f24e6a098c7f?"}
\ No newline at end of file
+{"mongoUri":"mongodb://127.0.0.1:58139/4e3c69b6-7035-4980-8689-537947773290?"}
```
Comment (gektin) [inline globalConfig.json:1]: Add a new line at the end of the file.

File: database/game_model.js [hunk 2]
```diff
@@ -26,7 +26,11 @@ const Game = new mongoose.model('Game', gameSchema);
...
 const getSingleGame = async (gameId) => {
   try {
-    return await Game.findOne({gameId}).exec();
+    const foundGame = await Game.findOne({gameId}).exec();
+    if (foundGame.length === 0) {
+      throw err;
+    }
+    return foundGame;
   } catch (error) {
     throw error;
```
Comment (gektin) [inline database/game_model.js:35]: Is this error being handled somewhere? I.e., is there somewhere this will be caught, or what happens if this line of code gets run?

File: server/routes/__tests__/games.test.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    const res = await request(app).get('/games/single/301');
+    expect(res.statusCode).toBe(400);
+    done();
+  });
+
+  it('should have a status code of 200 for multiple valid game ID\'s', async (done) => {
+    const res = await request(app).get('/games/multiple?tier1=1&tier1=2&tier2=3&tier2=4');
+    expect(res.statusCode).toBe(200);
+    done();
+  });
```
Comment (gektin) [inline server/routes/__tests__/games.test.js:22]: This is awesome functionality for your routes! As someone who will be calling them, this is super convenient.

File: server/routes/__tests__/games.test.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+const app = require('../../app');
+const request = require('supertest');
+
+describe('API routes', () => {
+
```
Comment (gektin) [inline server/routes/__tests__/games.test.js:5]: Do you want to test the information you're getting back from the routes as well? Checking the headers is helpful, but the real value here is the data coming back, right?

File: server/routes/games.js [hunk 5]
```diff
@@ -10,12 +9,16 @@ router.use((err, req, res, next) => {
 
 router.route('/single/:gameId').get( async (req, res, next) => {
   const gameId = req.params.gameId;
-  try {
-    const getSingleGame = await db.getSingleGame(gameId);
-    res.json(getSingleGame);
-  } catch (err) {
-    res.status(500).send('No game found');
-    next(err);
+  if (gameId < 1 || gameId > 300) {
```
Comment (gektin) [inline server/routes/games.js:12]: Do you know how this will handle a `gameId` of `15.5`? I've been using `parseInt()` prophylactically to avoid confronting the actual question under time constraint.

File: server/routes/games.js [hunk 6]
```diff
@@ -24,12 +27,17 @@ router.route('/multiple/').get( async (req, res) => {
...
     (async function getGames(gameIdsObj) {
       let gamesArray = [];
+      let invalidGameIds = [];
       for (const key in gameIdsObj) {
         const gameIds = gameIdsObj[key];
         if (gameIds) {
           if (Array.isArray(gameIds)) {
             for (const id of gameIds) {
-              gamesArray.push(db.getSingleGame(id));
+              if (id < 1 || id > 300) {
```
Comment (gektin) [inline server/routes/games.js:36]: Same comment as above.

File: server/routes/games.js [hunk 7]
```diff
@@ -24,12 +27,17 @@ router.route('/multiple/').get( async (req, res) => {
...
+                invalidGameIds.push(id);
+              } else {
+                gamesArray.push(db.getSingleGame(id));
+              }
             }
           }
           else {
```
Comment (gektin) [inline server/routes/games.js:43]: Start the `else` on the line above after the closing `}` for the `if` for readability.

File: server/routes/games.js [hunk 8]
```diff
@@ -38,10 +46,14 @@ router.route('/multiple/').get( async (req, res) => {
         }
       }
       const games = await Promise.all(gamesArray);
-      res.send(games);
+      if (invalidGameIds.length > 0) {
+        res.status(400).send('Invalid game ID(s), please enter valid ID(s)');
+      } else {
+        res.status(200).send(games);
+      }
     })(multipleGameIds)
```
Comment (gektin) [inline server/routes/games.js:54]: Out of curiosity, why is this function being invoked?

**Review by gektin — COMMENTED** [review]
On the whole, this looks good. There's some ways to expand your tests that I've added in line, along with some stylistic comments.

If you're confident you know what happens when line 35 of database/game_model.js runs and how you're dealing with floating point IDs, you can feel free to merge this in.

File: globalConfig.json (hunk 1 above)
Comment (redperezz) [reply globalConfig.json:1]: Done!

File: database/game_model.js (hunk 2 above)
Comment (redperezz) [reply database/game_model.js:35]: So when that error gets thrown on line 35 I believe it will get handled in server/app.js when a request is made. I will double check but in my head line 31 will be handled by that 404 in server/app.js and line 35 will be handled by that 500 in server/app.js

File: server/routes/__tests__/games.test.js (hunk 3 above)
Comment (redperezz) [reply server/routes/__tests__/games.test.js:22]: Awesome I'm glad!

File: server/routes/__tests__/games.test.js (hunk 4 above)
Comment (redperezz) [reply server/routes/__tests__/games.test.js:5]: Yes for sure! There are more tests coming soon

File: server/routes/games.js (hunk 5 above)
Comment (redperezz) [reply server/routes/games.js:12]: Wow great insight I haven't even that of that edge case, what an evil user to request half a game haha, I'll fix it!

File: server/routes/games.js (hunk 6 above)
Comment (redperezz) [reply server/routes/games.js:36]: Got it!

File: server/routes/games.js (hunk 7 above)
Comment (redperezz) [reply server/routes/games.js:43]: Nice eye, will fix!

File: server/routes/games.js (hunk 8 above)
Comment (redperezz) [reply server/routes/games.js:54]: I wasn't comfortable was async/await since I haven't used it much and this was what worked for me in my head lol
